pkg/middlewares/auth/basic_auth.go

```diff
@@ -65,6 +65,14 @@ func (b *basicAuth) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	if username := b.auth.CheckAuth(req); username == "" {
 		logger.Debug("Authentication failed")
 		tracing.SetErrorWithEvent(req, "Authentication failed")
+
+		if username, _, ok := req.BasicAuth(); ok {
```

Comment:
It is a bit silly and wasteful that we call req.BasicAuth() another time, whereas we have already computed the result in CheckAuth. Maybe L65 we should stop using CheckAuth, so we don't lose the result of req.BasicAuth (i.e. the username). Or maybe we could change the signature of CheckAuth so that it returns the username even when it does not match.
WDYT?

Reply:
I agree, it was bothering me also.
Changing the signature is breaking some other usages, if it also seems fair to you to call BasicAuth and check the secret in place of L65, i would go for that.

Reply:
yes, let's try that and see if it does not look too ugly.